### PR TITLE
fix: overly restrictive types

### PR DESCRIPTION
**Pull request by christianbundy**
Problem: Our types are overly restrictive, which means that the compiler warns us against things that aren't problems.

Solution: Fix Container, GridRow, GridItem, and Gutter.

### CODE REVIEW

File: packages/ui/src/Typography/index.tsx [hunk 1]
```diff
@@ -13,16 +13,15 @@ import {
...
 
-interface BoxProps extends SpaceProps, TypographyProps, LayoutProps, ColorProps {}
-
 const defaults = {
   fontFamily: theme.fonts.roboto,
   letterSpacing: '0.6px',
 }
 
 const sharedSystemProps = compose(space, layout, typography, color)
+interface SharedProps extends SpaceProps, TypographyProps, LayoutProps, ColorProps {}
```
Comment (christianbundy) [inline packages/ui/src/Typography/index.tsx:22]: This was named `BoxProps`, but it isn't.

**Comment by christianbundy** [comment]
CI seems to be failing because it wants to publish to netlify 🤦

**Review by wjohnson-truework — APPROVED** [review]
Looks good to me. Why remove the storybook deploy?

Storybook peer dependencies are keeping us on React 16 btw from what I can tell.

**Comment by christianbundy** [comment]
> CI seems to be failing because it wants to publish to netlify 🤦

**Pull request merged**